symbol lookup error

After compiling successfully with 0.27 and LLVM 4.0.0 (Ubuntu on the WSL) running the executable gives this error:
```./game_server: symbol lookup error: ./game_server: undefined symbol: errno, version Object::to_s<String::Builder>:String::Builder```.
The specific method referenced can change between computers (computers that were able to run the executables before this error appeared).

Unfortunately I can't provide code to reproduce because the program has over 150k lines. I'm trying to comment out code in the hopes of narrowing down where this comes from.

The problem doesn't happen if I compile with ```--single-module``` or in MacOS.

It's not much to go on but does anyone have an inkling of where this could be coming from?

It fails:

`/usr/bin/ld.lld: error: unknown argument: --push-state`
`/usr/bin/ld.lld: error: unknown argument: --pop-state`
`/usr/bin/ld.lld: error: unknown argument: --push-state`
`/usr/bin/ld.lld: error: unknown argument: --pop-state`
`collect2: error: ld returned 1 exit status`

So I will rebuild 0.32.1-build images with https://github.com/crystal-lang/crystal-dist/pull/8/files

And after that the following changes to the Makefile to build the specs https://github.com/crystal-lang/crystal/compare/ci/use-lld using lld

If `lld` is not installed the following error will be shown

```
CC="cc -fuse-ld=lld"  ./bin/crystal build  --exclude-warnings spec/std --exclude-warnings spec/compiler -o .build/std_spec spec/std_spec.cr
collect2: fatal error: cannot find 'ld'
compilation terminated.
Error: execution of command failed with code: 1: `cc -fuse-ld=lld "${@}" -o '/root/.cache/crystal/src-src-ecr-process.cr/macro_run'  -rdynamic  -lpcre -lm /usr/bin/../lib/crystal/lib/libgc.a -lpthread /src/src/ext/libcrystal.a -levent -lrt -ldl -L/usr/bin/../lib/crystal/lib -L/usr/lib -L/usr/local/lib`
Makefile:125: recipe for target '.build/std_spec' failed
```

In osx if `-fuse-ld=lld` is used the following error is shown, so the changes in the Makefile are applied on Linux only.

```
clang: warning: argument unused during compilation: '-fuse-ld=lld' [-Wunused-command-line-argument]
ld64.lld: warning: -sdk_version is required when emitting min version load command.  Setting sdk version to match provided min version
ld64.lld: error: Unable to find library for -lpthread
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Sound good?

I've also updated the https://github.com/crystal-lang/crystal/wiki/All-required-libraries#ubuntu

--- a/Makefile
+++ b/Makefile
@@ -65,6 +65,10 @@ DEPS = $(LLVM_EXT_OBJ) $(LIB_CRYSTAL_TARGET)
 CFLAGS += -fPIC $(if $(debug),-g -O0)
 CXXFLAGS += $(if $(debug),-g -O0)
 
+ifeq ($(shell command -v ld.lld >/dev/null && uname -s),Linux)
+  EXPORT_CC ?= CC="cc -fuse-ld=lld"
+endif
+
 ifeq (${LLVM_CONFIG},)
   $(error Could not locate llvm-config, make sure it is installed and in your PATH, or set LLVM_CONFIG)
 else
@@ -118,15 +122,15 @@ libcrystal: $(LIB_CRYSTAL_TARGET)
 
 $(O)/all_spec: $(DEPS) $(SOURCES) $(SPEC_SOURCES)
 	@mkdir -p $(O)
-	$(BUILD_PATH) ./bin/crystal build $(FLAGS) $(SPEC_WARNINGS_OFF) -o $@ spec/all_spec.cr
+	$(EXPORT_CC) $(BUILD_PATH) ./bin/crystal build $(FLAGS) $(SPEC_WARNINGS_OFF) -o $@ spec/all_spec.cr
 
 $(O)/std_spec: $(DEPS) $(SOURCES) $(SPEC_SOURCES)
 	@mkdir -p $(O)
-	$(BUILD_PATH) ./bin/crystal build $(FLAGS) $(SPEC_WARNINGS_OFF) -o $@ spec/std_spec.cr
+	$(EXPORT_CC) $(BUILD_PATH) ./bin/crystal build $(FLAGS) $(SPEC_WARNINGS_OFF) -o $@ spec/std_spec.cr
 
 $(O)/compiler_spec: $(DEPS) $(SOURCES) $(SPEC_SOURCES)
 	@mkdir -p $(O)
-	$(BUILD_PATH) ./bin/crystal build $(FLAGS) $(SPEC_WARNINGS_OFF) -o $@ spec/compiler_spec.cr
+	$(EXPORT_CC) $(BUILD_PATH) ./bin/crystal build $(FLAGS) $(SPEC_WARNINGS_OFF) -o $@ spec/compiler_spec.cr
 
 $(O)/crystal: $(DEPS) $(SOURCES)
 	@mkdir -p $(O)

